packages/amplify_auth_cognito/test/amplify_auth_cognito_stream_controller_test.dart

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import 'dart:async';
+import 'package:amplify_auth_cognito/amplify_auth_cognito_stream_controller.dart';
+import 'package:amplify_auth_cognito/src/CognitoHubEvents/AuthHubEvent.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:amplify_auth_cognito/amplify_auth_cognito.dart';
+import 'package:amplify_flutter/amplify.dart';
+import 'utils/get_json_from_file.dart';
+
+void main() {
+  const MethodChannel authChannel = MethodChannel('com.amazonaws.amplify/auth_cognito');
+  const String channelName = 'com.amazonaws.amplify/auth_cognito_events';
+
+  AmplifyAuthCognito auth = AmplifyAuthCognito();
+
+  TestWidgetsFlutterBinding.ensureInitialized();
+  
+
+  setUpAll(() async {
+    authChannel.setMockMethodCallHandler((MethodCall methodCall) async {
+      return true;
+    });
+  });
+
+  tearDown(() {
+    ServicesBinding.instance.defaultBinaryMessenger.setMockMessageHandler(channelName, null);
+  });
+
+  handler(event) {
+    ServicesBinding.instance.defaultBinaryMessenger.handlePlatformMessage(
+      channelName,
+      event,
+      (ByteData reply) {},
+    );
+  }
+
+  test('Amplify.Hub.listen with auth channel returns a StreamSubscription', () async {
+    await Amplify.addPlugin(auth);
+    var sub = Amplify.Hub.listen([HubChannel.Auth], (msg) {});
+    expect(sub, isInstanceOf<StreamSubscription>());
+    await sub.cancel();
+  });
```

Comment:
Can we move this test to `amplify_flutter` package to remove dependency on amplify_flutter on auth. Amplify_flutter should be our top level package.

Reply:
Done.